app/controllers/nodes_controller.rb

```diff
@@ -5,6 +5,16 @@
 # NodesController is responsible for everything related to nodes: showing
 # information on nodes, deleting them, etc.
 class NodesController < ApplicationController
+  rescue_from Minion::NonExistingMinion do |e|
```

Comment:
Nitpick: could you change this with:

```ruby
rescue_from Minion::NonExistingMinion, :minion_not_found
```

And then define a private method `minion_not_found` that does what you wrote ? I think that writing the whole thing in a lambda is confusing in a controller.